specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2021-07-01/machineLearningServices.json

```diff
@@ -0,0 +1,4768 @@
+              }
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "VirtualMachine"
+    },
+    "HDInsight": {
+      "description": "A HDInsight compute.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Compute"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "type": "object",
+              "properties": {
+                "sshPort": {
+                  "description": "Port open for ssh connections on the master node of the cluster.",
+                  "type": "integer",
+                  "format": "int32"
+                },
+                "address": {
+                  "description": "Public IP address of the master node of the cluster.",
+                  "type": "string"
+                },
+                "administratorAccount": {
+                  "description": "Admin credentials for master node of the cluster",
+                  "$ref": "#/definitions/VirtualMachineSshCredentials"
+                }
+              }
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "HDInsight"
+    },
+    "DataFactory": {
+      "description": "A DataFactory compute.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Compute"
+        }
+      ],
+      "x-ms-discriminator-value": "DataFactory"
+    },
+    "Databricks": {
+      "description": "A DataFactory compute.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Compute"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "type": "object",
+              "properties": {
+                "databricksAccessToken": {
+                  "description": "Databricks access token",
+                  "type": "string"
+                },
+                "workspaceUrl": {
+                  "description": "Workspace Url",
+                  "type": "string"
+                }
+              }
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "Databricks"
+    },
+    "DataLakeAnalytics": {
+      "description": "A DataLakeAnalytics compute.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Compute"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "type": "object",
+              "properties": {
+                "dataLakeStoreAccountName": {
+                  "description": "DataLake Store Account Name",
+                  "type": "string"
+                }
+              }
+            }
+          }
+        }
+      ],
+      "x-ms-discriminator-value": "DataLakeAnalytics"
+    },
+    "SynapseSpark": {
+      "description": "A SynapseSpark compute.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Compute"
+        },
+        {
+          "$ref": "#/definitions/SynapseSparkPoolProperties"
```

Comment:
This is a potential source of `DuplicatedSchema` in the modeler. instead of using allOf to inject the properties, could we just define those properties in this `SynapseSpark` model and remove the definition of `SynapseSparkPoolProperties`?

Reply:
Sorry, I would like to make sure I understand you. Are you saying I should unwrap `SynapseSparkPoolProperties` to something like `AKS`
![image](https://user-images.githubusercontent.com/7452880/124188080-9fade500-da73-11eb-8ea4-eb51df8097ad.png)